packages/loot-core/src/server/budget/statements.ts

```diff
@@ -0,0 +1,47 @@
+import * as db from '../db';
+import { Schedule } from '../db/types';
+
+import { GOAL_PREFIX, TEMPLATE_PREFIX } from './template-notes';
+
+/* eslint-disable rulesdir/typography */
+export async function resetCategoryGoalDefsWithNoTemplates(): Promise<void> {
+  await db.run(
+    `
+      UPDATE categories
+      SET goal_def = NULL
```

Comment:
I maybe forgot to add it before, but if the goal is getting removed, long_goal should also be set to null.

Reply:
Is that the long_goal in the `reflect_budgets` table?

Reply:
its in both budget tables

Reply:
Ahh I see. I saw the `resetCategoryTargets` gets called a bit later when the goals are processed. In that function it does the reset of all the budget categories as you mentioned. I was planning to refactor that function in its own PR as it will be quite a biggie. Do you think the long_goals should be reset regardless?

Reply:
its not a big deal if you would rather look at it later.  Plus if its not causing trouble then we can let it lie

Reply:
I can tackle it when I work on the `processGoals` function, and hopefully we can just do one reset when applying the goals regardless if it comes from notes or the future UI